fraud/testing.go

```diff
@@ -66,3 +70,54 @@ func generateByzantineError(
 	_, err := rtrv.Retrieve(ctx, faultHeader.DAH)
 	return faultHeader, err
 }
+
+const (
+	mockProofType ProofType = "mockProof"
+)
+
+type mockProof struct {
+	Valid bool
+}
+
+func newValidProof() *mockProof {
+	return newMockProof(true)
+}
+
+func newInvalidProof() *mockProof {
+	return newMockProof(false)
+}
+
+func newMockProof(valid bool) *mockProof {
+	p := &mockProof{valid}
+	if _, ok := defaultUnmarshalers[p.Type()]; !ok {
```

Comment:
test should probably also use the lock to prevent races/flakeyness, no?

Reply:
we run test synchronously.